File: parse_logs.py
import glob
import csv
import re

# 1) Find all relaxed_*.log files
files = sorted(glob.glob("logs_pdlp_1e-06/relaxed_*.log"))

# 2) Prepare CSV output
with open("summary.csv", "w", newline="") as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(["filename", "optimal_objective", "time"])  # header

    for file in files:
        with open(file, "r") as f:
            content = f.read()

            # 3) Regex to extract Optimal objective
            obj_match = re.search(r"Objective = ([\-\d\.]+)", content)

            # 4) Regex to extract Time (last Time: value in seconds)
            time_match = re.search(r"Time: ([\d\.]+)s", content)

            objective = obj_match.group(1) if obj_match else ""
            time = time_match.group(1) if time_match else ""

            # 5) Write row
            writer.writerow([file, objective, time])

print("CSV created successfully: summary.csv")

File: parse_logs.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <regex>
#include <filesystem>

namespace fs = std::filesystem;
using namespace std;

int main() {
    vector<string> files;

    for (const auto& entry : fs::directory_iterator("logs_pdlp_1e-06")) {
        if (entry.path().string().find("relaxed_") != string::npos && 
            entry.path().extension() == ".log") {
            files.push_back(entry.path().string());
        }
    }

    sort(files.begin(), files.end());

    ofstream csv("summary.csv");
    csv << "filename,optimal_objective,time\n";

    regex objRegex(R"(Objective = ([\-\d\.]+))");
    regex timeRegex(R"(Time: ([\d\.]+)s)");

    for (const auto& file : files) {
        ifstream f(file);
        string content((istreambuf_iterator<char>(f)), {});
        smatch objMatch, timeMatch;

        string objective = "";
        string time = "";

        if (regex_search(content, objMatch, objRegex)) {
            objective = objMatch[1];
        }
        if (regex_search(content, timeMatch, timeRegex)) {
            time = timeMatch[1];
        }

        csv << file << "," << objective << "," << time << "\n";
    }

    cout << "CSV created: summary.csv" << endl;
    return 0;
}
